src/commands/navigation.rs

```diff
@@ -192,11 +198,33 @@ pub fn checkout(effects: &Effects, git_run_info: &GitRunInfo) -> eyre::Result<is
     let graph = make_smartlog_graph(effects, &repo, &dag, &event_replayer, event_cursor, true)?;
 
     match prompt_select_commit(graph.get_commits(), None)? {
-        Some(oid) => {
-            let result = git_run_info.run(effects, None, &["checkout", &oid.to_string()])?;
-            smartlog(effects, &Default::default())?;
-            Ok(result)
-        }
+        Some(oid) => checkout_commit(effects, git_run_info, oid),
         None => Ok(1),
     }
 }
+
+fn checkout_commit<T>(
+    effects: &Effects,
+    git_run_info: &GitRunInfo,
+    target: T,
+) -> eyre::Result<isize>
+where
+    T: ToString,
```

Comment:
You call `to_string` more than once in this function, so you're allocating twice unnecessarily. It would be better to accept a `&str` and have the caller allocate the `String` only once (or zero times if it's a static string!).

As a side note, you can write this trait bound in any of the following ways:

- `foo(s: impl ToString)`
- `foo<T: ToString>(s: T)`
- `foo<T>(s: T) where T: ToString`

For this project, you should prefer them in that order, and use the more complex versions only when necessary to express more complicated constraints.

Reply:
Done. (And I'll keep the trait style in mind for the future)